controllers/torrent/files.go

```diff
@@ -0,0 +1,78 @@
+package torrentController
+
+import (
+	"html/template"
+	"encoding/hex"
+	"net/http"
+	"strings"
+	"strconv"
+
+	"github.com/NyaaPantsu/nyaa/models/torrents"
+	"github.com/NyaaPantsu/nyaa/models"
+	"github.com/NyaaPantsu/nyaa/templates"
+	"github.com/NyaaPantsu/nyaa/utils/format"
+	"github.com/NyaaPantsu/nyaa/utils/filelist"
+	"github.com/Stephen304/goscrape"
+	"github.com/gin-gonic/gin"
+)
+
+func GetFilesHandler(c *gin.Context) {
+	id, _ := strconv.ParseInt(c.Param("id"), 10, 32)
+	torrent, err := torrents.FindByID(uint(id))	
+
+	if err != nil {
+		c.Status(http.StatusNotFound)
+		return
+	}
+	
+	
+	if len(torrent.FileList) == 0 {
+		var blankScrape models.Scrape
+		ScrapeFiles(format.InfoHashToMagnet(strings.TrimSpace(torrent.Hash), torrent.Name, GetTorrentTrackers(torrent)...), torrent, blankScrape, true)
+	}
+	
+	folder := filelist.FileListToFolder(torrent.FileList, "root")
+	templates.TorrentFileList(c, torrent.ToJSON(), folder)
+}
+
+// ScrapeFiles : Scrape torrent files
+func ScrapeFiles(magnet string, torrent *models.Torrent, currentStats models.Scrape, statsExists bool) (error, []FileJSON) {
+	if client == nil {
+		err := initClient()
+		if err != nil {
+			return err, []FileJSON{}
+		}
+	}
+	
+	t, _ := client.AddMagnet(magnet)
+	<-t.GotInfo()

```

Comment:
If it's only for the filelist, you should detect before if you have the torrent file and then if you have it already, load the files from it (since you won't have to wait to connect to peers to have them). You can check how I access to a torrent file information in the upload process here with `tfile` the content of the torrent file 
https://github.com/NyaaPantsu/nyaa/blob/860f9c8eb1c001546402bd9a36c0fda9d97feb00/utils/validator/torrent/functions.go#L195

In the opposite, if you don't have it, you should save it after the scrapping so you save ressource power